renovate-base.json5

```diff
@@ -62,6 +62,20 @@
         // The packageRules are overrided/merged, therefore this rule needs to set automerge to false explicitly.
         "automerge": false,
     },
+    {
+        // go.mod golang-version is not updated by default, check <https://github.com/renovatebot/renovate/issues/16715>.
+        // Enable golang version bumps in 'go.mod' but disable auto-merge.
+        "matchDatasources": ["golang-version"],
+        // We always merge patch updates for golang.
+        "matchUpdateTypes": ["patch"],
+        // Exclude any dependencies which are pre-1.0.0 because those can make breaking changes at any time
+        // according to the SemVer spec.
+        "matchCurrentVersion": "!/^(0|v0)/",
```

Comment:
I guess this doesn't apply to Go versions since they are all already `v1` as of now, but good to have it around nonetheless 👌